File: apps/dot-net-sdk/Sitecrawl/Exceptions/AuthenticationException.cs
namespace Sitecrawl.Exceptions;

/// <summary>
/// Thrown when the API returns a 401 Unauthorized response.
/// </summary>
public class AuthenticationException : SitecrawlException
{
    public AuthenticationException(string message, string? errorCode = null, object? details = null)
        : base(message, 401, errorCode, details)
    {
    }
}

File: apps/java-sdk/src/main/java/com/sitecrawl/errors/AuthenticationException.java
package com.sitecrawl.errors;

/**
 * Thrown when the API returns a 401 Unauthorized response.
 */
public class AuthenticationException extends SitecrawlException {

    public AuthenticationException(String message) {
        super(message, 401);
    }

    public AuthenticationException(String message, String errorCode, Object details) {
        super(message, 401, errorCode, details);
    }
}
